pandas/core/internals/blocks.py

```diff
@@ -1912,7 +1912,9 @@ def get_block_type(values, dtype: Dtype | None = None):
         cls = ExtensionBlock
     elif isinstance(dtype, CategoricalDtype):
         cls = CategoricalBlock
-    elif vtype is Timestamp:
+    # error: Non-overlapping identity check (left operand type: "Type[generic]",
+    # right operand type: "Type[Timestamp]")
+    elif vtype is Timestamp:  # type: ignore[comparison-overlap]
```

Comment:
There's a cast on L1895 that looks iffy. remove it and this ignore is not needed.

Reply:
i've merged master locally so the line numbers differ.